Regular Expressions/RegExp (Basic).sublime-syntax

```diff
@@ -90,84 +86,123 @@ contexts:
       pop: 1
 
   groups:
-    - include: group-comments
     - match: \(
       scope: keyword.control.group.regexp
-      push: group-start
+      push: [group-body-placeholder, maybe-unexpected-quantifiers, group-start]
 
   groups-extended:
-    - include: group-comments
     - match: \(
       scope: keyword.control.group.regexp
-      push: group-start-extended
+      push: [group-body-extended-placeholder, maybe-unexpected-quantifiers, group-start-extended]
+
+  group-body-placeholder:
+    - match: ''
+      set: group-body
+
+  group-body-extended-placeholder:
+    - match: ''
+      set: group-body-extended
 
   group-start:
     - meta_scope: meta.group.regexp
-    - match: \?(<[=!]|[>=:!])
-      scope: constant.other.assertion.regexp
-      set: [group-body, maybe-unexpected-quantifiers]
-    # Activates 'x' mode
-    - match: (\?[ims]*x[ixms]*(?:-[ims]+)?)(\))
-      captures:
-        1: storage.modifier.mode.regexp
-        2: keyword.control.group.regexp
-      set: [base-group-extended, maybe-unexpected-quantifiers]
-    # Groups with 'x' mode
-    - match: '\?[ims]*x[ixms]*(?:-[ims]+)?:'
-      scope: storage.modifier.mode.regexp
-      set: [group-body-extended, maybe-unexpected-quantifiers]
-    # Other modifiers
-    - match: '\?(?:[ixms]*-)?[ixms]+:'
-      scope: storage.modifier.mode.regexp
-      set: [group-body, maybe-unexpected-quantifiers]
     - include: group-start-common
-    - match: ''
-      set: [group-body, maybe-unexpected-quantifiers]
 
   group-start-extended:
     - meta_scope: meta.group.extended.regexp
+    - include: group-start-common
+
+  group-start-common:
     - match: \?(<[=!]|[>=:!])
       scope: constant.other.assertion.regexp
+      pop: 1
+    # Groups which activate 'x' mode
+    - match: '{{activate_x_mode}}:'
+      scope: storage.modifier.mode.regexp
+      pop: 3
       set: [group-body-extended, maybe-unexpected-quantifiers]
-    # Deactivates 'x' mode
-    - match: (\?[ims]*-[ims]*x[imxs]*)(\))
-      captures:
-        1: storage.modifier.mode.regexp
-        2: keyword.control.group.regexp
-      set: [base-group, maybe-unexpected-quantifiers]
-    # Groups without 'x' mode
-    - match: '\?[ims]*-[ims]*x[imxs]*:'
+    # Groups which deactivate 'x' mode
+    - match: '{{deactivate_x_mode}}:'
       scope: storage.modifier.mode.regexp
+      pop: 3
       set: [group-body, maybe-unexpected-quantifiers]
     # Other modifiers
-    - match: '\?(?:[ixms]*-)?[ims]+:'
+    - match: '\?(?:[ixms]*-)?[ixms]+:'
       scope: storage.modifier.mode.regexp
-      set: [group-body-extended, maybe-unexpected-quantifiers]
-    - include: group-start-common
-    - match: ''
-      set: [group-body-extended, maybe-unexpected-quantifiers]
-
-  group-start-common:
+      pop: 2
     # Other modifiers (must come after other modifier matches)
     - match: (\?(?:[ixms]*-)?[ixms]+)(\))
       captures:
         1: storage.modifier.mode.regexp
         2: keyword.control.group.regexp
+      pop: 2
```

Comment:
We propably don't want to end up in `group-body` if a closing paren is already matched?

```suggestion
      pop: 3
```